system/core/RequestContext.php

```diff
@@ -120,6 +124,62 @@ public function getDictionary() {
 			return $this->dictionary;
 		}
 
+		/*
+			Function: getCsrfToken
+
+			Returns the CsrfToken to be used for all
+			subsequent POST requests. If there is no token
+			yet, it will be generated automatically.
+		*/
+		public function getCsrfToken() {
```

Comment:
namingwise I would expect get to be just a getter (not meddling with state) the regeneration breaks that expectation. Maybe getOrGenerateCsrfToken / retrieveCsrfToken?

Reply:
Good point, fixed.